tests/test_models.py

```diff
@@ -83,6 +89,83 @@ def test_model_singlepsr_noise_dip_cusp(nodmx_psrs,caplog):
     x0 = {pname:p.sample() for pname,p in zip(m.param_names, m.params)}
     m.get_lnlikelihood(x0)
 
+def test_model_singlepsr_noise_dm_nondiag(nodmx_psrs,caplog):
+    # caplog.set_level(logging.CRITICAL)
+    ii = 1
+    mn=models.model_singlepsr_noise(nodmx_psrs[ii], dm_var=True,
+                                dm_type='gp', dmgp_kernel ='nondiag',
+                                dm_nondiag_kernel ='dmx_like')
+    assert hasattr(mn,'get_lnlikelihood')
+    x0 = {pname:p.sample() for pname,p in zip(mn.param_names, mn.params)}
+    assert psr_name[ii]+'_dm_gp_log10_sigma' in mn.param_names
```

Comment:
We want to test a known input against a known output.  The test should compare against a static object.  The parameter name (including which pulsar we're testing) should be hard coded, not dynamically determined